lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -453,6 +455,15 @@ def _extract_collection_from_git(repo_url, coll_ver, b_path):
             proc_err,
         )
 
+    has_submodules = (pathlib.Path(to_native(b_checkout_path)) / '.gitmodules').is_file()
+    with_submodule_cli_request = context.CLIARGS['git_init_submodules']
```

Comment:
This value should be passed through the cascade of calls, not accessed directly. It's evident in the failing unit tests how this causes abstraction leaks.

Reply:
it's complicated. i think it's better to use context.CLIARGS.get(name, default) as it's done in https://github.com/ansible/ansible/blob/devel/lib/ansible/galaxy/role.py#L345

this is the call sequence:

![image](https://user-images.githubusercontent.com/4615212/187756591-272212aa-8c30-4f44-8e05-9071b48c67f8.png)

Reply:
I think it's unacceptable to do this. I wrote this code as well as most things under `lib/ansible/galaxy/dependency_resolution/` and `lib/ansible/galaxy/collection/` with a clear intent to keep it as self-contained as possible. This is why, if you look at the code in the module, it doesn't access magical global contexts.

`lib/ansible/galaxy/role.py` has existed like this historically, which is the only reason it still has these bits of code smell in it. Besides, it's not a fair comparison since you're pointing at a module in a different importable package, on a different abstraction layer. There's no such examples under `lib/ansible/galaxy/collection/*` where you actually do modifications.

OTOH, if you look closer at https://github.com/danielino/ansible/blob/devel/lib/ansible/cli/galaxy.py#L1017-L1021, you'll see that all CLI arguments are inspected and translated there, in that module. And only things necessary for the function from `lib/ansible/galaxy/collection/` are actually passed there, in the correct form.

Also, this value wouldn't need to be passed to every method call in that stack but probably just added to the initializer of `ConcreteArtifactsManager` and stored as an instance attribute and then, accessed via `self` later following the structure that is already in place. Adding an extra approach that doesn't follow literally everything else would make the code more entangled, hard to follow, unmaintainable. We don't want that.